pkg/config/analysis/analyzers/util/service_lookup.go

```diff
@@ -61,6 +62,17 @@ func InitServiceEntryHostMap(ctx analysis.Context) map[ScopedFqdn]*v1alpha3.Serv
 		return true
 	})
 
+	// use meshConfig.trustDomain changed the default domain
+	// todo: replace by `values.global.proxy.clusterDomain`
```

Comment:
Based on the description above( why use ** trusDomain** ??), I believe the current solution is only a preliminary workaround. A more optimal solution is needed and requires further discussion.

And I'm not sure how the community plans and designs this aspect, and whether there's a need for unification. Additionally, I think it would be helpful to include some usage documentation, as many people might encounter the same issue.

@zirain @howardjohn @hzxuzhonghu @hanxiaop